src/wovnio/wovnphp/custom_domain/CustomDomainLangs.php

```diff
@@ -43,9 +66,52 @@ public function getCustomDomainLangByUrl($url)
         return count($results) <= 0 ? null : array_shift($results);
     }
 
+    public function toHtmlSwapperHash()
+    {
+        $result = array();
+        foreach ($this->customDomainLangs as $lang) {
+            $result[$lang->getHostAndPathWithoutTrailingSlash()] = $lang->getLang();
+        }
+        return $result;
+    }
+
+    /**
+     * Returns the computed (virtual) uri representation for a given physical location uri.
+     * Used when communicating with html-swapper.
+     *
+     * @param $physicalUri string the current uri - pointing to physical location of current lang
+     * @param $lang string lang code of the current uri
+     * @param $defaultLang string lang code of the default (source) language
+     * @return string|string[]|null
```

Comment:
Can the return value really be string or a string array? if so, this is not very ideal.

Reply:
Actually yes. If they caller pass in an array to $physicalUri the return value would be an array of strings. (not the intended use but it works that way.)

Reply:
Could add a type check for input params but I don't think it's worth it. We would be basically fighting PHP itself.

Reply:
i'm confused still. Isn't `$physicalUri` always a string?  what's an example of an array return value?

Reply:
Sorry I didn't phrase this clearly. Inside the usage of WOVN.php $physicalUri is never an array, it is always a string. But because inside `changeToNewCustomDomainLang`, which `computeSourceVirtualUrl` calls, there's a call to `preg_replace` that's capable of accepting and returning an array. And since PHP doesn't do type checking it makes the `computeSourceVirtualUrl` capable of returning an array as well. 

I'll just update the doc string to remove the string[] to make things clearer.

Reply:
Yeah, if preg_replace only ever returns string in the situations we use it in, just ignore the array case.